src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticStartAnalysisScope.cs

```diff
@@ -433,11 +427,9 @@ public override AnalyzerActions GetAnalyzerActions(DiagnosticAnalyzer analyzer)
     /// <summary>
     /// Scope for setting up analyzers for analyzing a symbol and its members.
     /// </summary>
-    internal sealed class HostSymbolStartAnalysisScope : HostAnalysisScope
+    internal sealed class HostSymbolStartAnalysisScope(DiagnosticAnalyzer analyzer)
```

Comment:
>HostSymbolStartAnalysisScope [](http://example.com/codeflow?start=26&length=28)

Should HostCodeBlockStartAnalysisScope and HostOperationBlockStartAnalysisScope also hold onto an analyzer so their Register* methods need not take in an analyzer for consistency?

Reply:
Which would make AnalyzerCodeBlockStartAnalysisContext and AnalyzerOperationBlockStartAnalysisContext not need to take in an analyzer directly, right?

Reply:
⏱️ I will look into this as soon as https://github.com/dotnet/arcade/issues/14761 is resolved. Until that time I can't really make progress in an arcade-enabled repository due to a wall of errors.

Reply:
Wanted to ping on this PR, as I think it's a good change and would like to see it move forward. Are you still blocked by the Arcade issue?

Reply:
➡️ These types are now updated for consistency